src/nsolid/nsolid_heap_snapshot.cc

```diff
@@ -38,6 +68,16 @@ int NSolidHeapSnapshot::GetHeapSnapshot(SharedEnvInst envinst,
   return status;
 }
 
+void NSolidHeapSnapshot::start_tracking_heapobjects(
+    SharedEnvInst envinst,
+    bool trackAllocations,
+    NSolidHeapSnapshot* snapshotter) {
+  ASSERT_EQ(snapshotter->is_tracking_heapobjects_, false);
+  envinst->isolate()->GetHeapProfiler()->StartTrackingHeapObjects(
+      trackAllocations);
+  snapshotter->is_tracking_heapobjects_ = true;
```

Comment:
Check the return value just in case the inspector is running a profiler itself

Reply:
Double-checked with Node.js and V8; nothing bad is going to happen.